w3af/plugins/crawl/open_api.py

```diff
@@ -292,6 +315,35 @@ def _analyze_current_path(self, fuzzable_request):
             self._spec_url_generator_current_path(fuzzable_request)
         )
 
+    def _has_custom_spec_location(self):
+        """
+        Checks if the plugin is configured to use a custom API specification
+        from a local file.
+
+        :return: True if the plugin is configured to read a custom API spec
+        """
+        return self._custom_spec_location != ''
+
+    def _analyze_custom_spec(self):
+        """
+        Loads a custom API specification from a local file, and try to parse it.
+
+        :return: None
+        """
+        if not self._first_run:
+            return
+        self._first_run = False
+
+        url = URL('file://%s' % os.path.abspath(self._custom_spec_location))
+
+        with open(self._custom_spec_location, 'r') as f:
+            custom_spec_as_string = f.read()
+
+        headers = Headers([('content-type', 'application/json')])
```

Comment:
Correct, both yaml and json are supported. It looks like the header is just ignored, or maybe it doesn't affect much parsing the spec because I tested it with yaml, and it worked well. I just followed your advice here

https://github.com/andresriancho/w3af/issues/15087#issuecomment-380089063

I'll check it it makes sense to set the header with correct type, or if it's fine to drop it.

Reply:
I see that `OpenAPI.can_parse()` checks Content-Type header but after some testing I don't think that `Content-Type: application/json` breaks processing an Open API spec in YAML format. Nevertheless, I think it would be better to specify a correct content type here at least to prevent further confusions.